Classes/Configuration/PhinxConfiguration.php

```diff
@@ -28,22 +32,49 @@ public function toArray(): array
             'environments' => [
                 'default_migration_table' => self::MIGRATION_TABLE_NAME,
                 'default_environment' => 'typo3',
-                'typo3' => [
-                    'adapter' => 'mysql',
-                    'host' => $connectionParameters['host'],
-                    'port' => $connectionParameters['port'],
-                    'user' => $connectionParameters['user'],
-                    'pass' => $connectionParameters['password'],
-                    'name' => $connectionParameters['dbname'],
-                    'unix_socket' => $connectionParameters['unix_socket'] ?? null,
-                    'charset' => $connectionParameters['charset'],
-                ],
+                'typo3' => $this->getEnvironment($connection),
             ],
         ];
     }
 
-    private function getConnectionParameters(): array
+    private function getEnvironment(Connection $connection): array
     {
-        return $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections']['Default'];
+        switch ($connection->getDatabasePlatform()->getName()) {
```

Comment:
Seems like we could put this directly in `$adapter` instead.

Also what's the reason from using a connection instead of the raw TYPO3 config?